### PR TITLE
DDBTEAM-346: Added RbDigitalBooks vendor

### PR DESCRIPTION
**Pull request by turegjorup**

### CODE REVIEW

File: src/Service/VendorService/RbDigital/DataConverter/RbDigitalPublicUrlConverter.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * @file
+ * Convert 'Iverse' urls from pointing to medium sized images to pointing large size.
+ */
+
+namespace App\Service\VendorService\RbDigital\DataConverter;
+
+/**
+ * Class IversePublicUrlConverter.
```
Comment (tuj) [inline src/Service/VendorService/RbDigital/DataConverter/RbDigitalPublicUrlConverter.php]: Should be RbDigitalPublicUrlConverter

File: src/Service/VendorService/RbDigital/DataConverter/RbDigitalPublicUrlConverter.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * @file
+ * Convert 'Iverse' urls from pointing to medium sized images to pointing large size.
```
Comment (tuj) [inline src/Service/VendorService/RbDigital/DataConverter/RbDigitalPublicUrlConverter.php]: pointing **to**

File: src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php [hunk 3]
```diff
@@ -0,0 +1,221 @@
+<?php
+
```
Comment (tuj) [inline src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php]: Missing @file comment

File: src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+use Doctrine\ORM\EntityManagerInterface;
+use League\Flysystem\FileNotFoundException;
+use League\Flysystem\Filesystem;
+use Psr\Cache\InvalidArgumentException;
+use Psr\Log\LoggerInterface;
+use Scriptotek\Marc\Collection;
+use Symfony\Component\Cache\Adapter\AdapterInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class RbDigitalBooksVendorService extends AbstractBaseVendorService
```
Comment (tuj) [inline src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php]: Missing class comment

File: src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+     * @param filesystem $ftp
+     *   Flysystem adapter for remote ftp server
+     * @param entityManagerInterface $entityManager
+     *   Doctrine entity manager
+     * @param loggerInterface $statsLogger
+     *   Logger object to send stats to ES
+     * @param AdapterInterface $cache
+     *   Cache adapter for the application
+     */
+    public function __construct(EventDispatcherInterface $eventDispatcher, Filesystem $local,
```
Comment (tuj) [inline src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php]: All function parameters on one line.

File: src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+    ];
+
+    private $local;
+    private $ftp;
+    private $cache;
+
+    /**
+     * RbDigitalVendorService constructor.
+     *
+     * @param eventDispatcherInterface $eventDispatcher
```
Comment (tuj) [inline src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php]: lowercase first letter of type should be uppercase for all param notations.

File: src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+
+        $this->progressFinish();
+
+        return VendorImportResultMessage::success($this->totalIsIdentifiers, $this->totalUpdated, $this->totalInserted, $this->totalDeleted);
+    }
+
+    /**
+     * Update local copy of vendors archive.
+     *
+     * @param string $mrcFile
```
Comment (tuj) [inline src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php]: Missing description.

File: src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+        $this->cache->save($remoteModifiedAtCache);
+
+        // @TODO Error handling for missing archive
+        return $this->local->put($mrcFile, $this->ftp->read($mrcFile));
+    }
+
+    /**
+     * Check if vendors archive has update.
+     *
+     * @param string $mrcFile
```
Comment (tuj) [inline src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php]: Missing description.

File: src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+            }
+        }
+
+        return $update;
+    }
+
+    /**
+     * Get cache key for the given filename.
+     *
+     * @param string $mrcFile
```
Comment (tuj) [inline src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php]: Missing description.

File: src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+     */
+    private function getCacheKey(string $mrcFile): string
+    {
+        $hash = md5($mrcFile);
+
+        return 'app.vendor.'.$this->getVendorId().$hash.'.remoteModifiedAt';
+    }
+
+    /**
+     * Set config from service from DB vendor object.
```
Comment (tuj) [inline src/Service/VendorService/RbDigital/RbDigitalBooksVendorService.php]: Set config **for** service

**Comment by turegjorup** [comment]
PR updated

**Pull request merged**